### PR TITLE
docker support with env variables support

### PR DESCRIPTION
**Pull request by maxxxdj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,19 @@ See [index.js](https://github.com/mifi/instauto/blob/master/index.js) for availa
 
 See [example.js](https://github.com/mifi/instauto/blob/master/example.js) for example of features
 
+## Docker compose by https://github.com/maxxxdj
+
+- We've added a Docker support for insta-auto!
```
Comment (mifi) [inline README.md:43]: The name of the project is `instauto`, so I'd prefer to keep that name instead of `insta-auto`

File: example.js [hunk 2]
```diff
@@ -7,21 +7,21 @@ const Instauto = require('instauto'); // eslint-disable-line import/no-unresolve
...
   cookiesPath: './cookies.json',
 
-  username: 'your-ig-username',
-  password: 'your-ig-password',
+  username: process.env.INSTAGRAM_USERNAME,
+  password: process.env.INSTAGRAM_PASSWORD,
 
   // Global limit that prevents follow or unfollows (total) to exceed this number over a sliding window of one hour:
-  maxFollowsPerHour: 20,
+  maxFollowsPerHour: process.env.MAX_FOLLOWS_PER_HOUR,
```
Comment (mifi) [inline example.js:14]: For all these parameters, can we instead default to the old value if environment is not set? Also need to convert most of these to `Number`.
e.g.:
```js
maxFollowsPerHour: process.env.MAX_FOLLOWS_PER_HOUR != null ? parseInt(process.env.MAX_FOLLOWS_PER_HOUR, 10) : 20
```

**Review by mifi — CHANGES_REQUESTED** [review]
Thanks. I left some feedback. I see there's a lot of changes to unrelated files. I think all changes to other files **except** Dockerfile, README.md, docker-compose.yaml and example.js can be removed, am I right?

**Comment by mifi** [comment]
thanks. I did some additional improvements. I still see that yarn.lock, package.json, index.js changes are part of this PR. I don't think they are needed? If not, could you revert those?

**Comment by maxxxdj** [comment]
> thanks. I did some additional improvements. I still see that yarn.lock, package.json, index.js changes are part of this PR. I don't think they are needed? If not, could you revert those?

I've tried with the old ones, but Dockerfile was failing to build.
image
I suggest to try to run it locally and if it's not a problem to leave them.
Cheers!

**Comment by mifi** [comment]
That’s odd. Maybe try yarn add instuto instead of npm install instauto

**Comment by maxxxdj** [comment]
> That’s odd. Maybe try yarn add instuto instead of npm install instauto

It worked! Great, thanks.
The only changes now are some IDE or whitespaces.
I've changed the docker container name to be the correct as your project and product - "instauto".

Cheers

**Comment by mifi** [comment]
thanks! i would appreciate if you also remove the whitespace changes. after that is solved i think this can be merged

**Comment by maxxxdj** [comment]
> thanks! i would appreciate if you also remove the whitespace changes. after that is solved i think this can be merged

done!

File: example.js [hunk 3]
```diff
@@ -7,21 +7,21 @@ const Instauto = require('instauto'); // eslint-disable-line import/no-unresolve
...
 
   // Global limit that prevents follow or unfollows (total) to exceed this number over a sliding window of one hour:
-  maxFollowsPerHour: 20,
+  maxFollowsPerHour: process.env.MAX_FOLLOWS_PER_HOUR != null ? parseInt(process.env.MAX_FOLLOWS_PER_HOUR, 10) : 20,
   // Global limit that prevents follow or unfollows (total) to exceed this number over a sliding window of one day:
-  maxFollowsPerDay: 150,
+  maxFollowsPerDay: process.env.MAX_FOLLOWS_PER_DAY != null ? parseInt(process.env.MAX_FOLLOWS_PER_DAY, 10) : 150,
   // (NOTE setting the above parameters too high will cause temp ban/throttle)
 
-  maxLikesPerDay: 50,
```
Comment (mifi) [inline example.js:19]: Any reason why you changed the default from 50 to 30?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+WORKDIR /app/instauto
+
+COPY package.json /app/instauto
+
+RUN yarn add instauto
+
+COPY . /app/instauto
+
+CMD ["yarn", "upgrade", "puppeteer:latest",  "instauto"]
```
Comment (mifi) [inline Dockerfile:14]: Why is this line needed?

**Comment by thegreenmilecornant** [comment]
Any news on this PR? Would love to use a simple docker compose to deploy instauto.

Tried building an image with the dockerfile (first time) - but failed.

**Comment by maxxxdj** [comment]
> Any news on this PR? Would love to use a simple docker compose to deploy instauto.
> 
> Tried building an image with the dockerfile (first time) - but failed.

Hey!
I'm not sure why the owner didn't merged this one.
It was running fine on my side and even I had it with env variables so I can run different accounts.
If you copied that file and trying it on the current master maybe won't work, due to changes in master branch.

Try to ping mifi

**Comment by mifi** [comment]
Hi. I left some code review questions that were not answered

**Comment by thegreenmilecornant** [comment]
> Hey! I'm not sure why the owner didn't merged this one. It was running fine on my side and even I had it with env variables so I can run different accounts. If you copied that file and trying it on the current master maybe won't work, due to changes in master branch.
> 
> Try to ping mifi

OK. Would you be so kind and look at the requested changes by @mifi and update the PR 😃 ?
Would love to deploy this in Docker!

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      dockerfile: ./Dockerfile
+    environment:
+      - INSTAGRAM_USERNAME=username
+      - INSTAGRAM_PASSWORD=password
+      - MAX_FOLLOWS_PER_HOUR=15
+      - MAX_FOLLOWS_PER_DAY=70
+      - MAX_LIKES_PER_DAY=30
+      - FOLLOW_USER_RATIO_MIN=0.2
+      - FOLLOW_USER_RATIO_MAX=4.0
+      - USERS_TO_FOLLOW=users
```
Comment (mifi) [inline docker-compose.yaml:15]:
```suggestion
      - USERS_TO_FOLLOW=everchanginghorizon,sam_kolder
```

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN apt-get update
+RUN apt-get install -y gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libnss3 lsb-release xdg-utils wget
+
+WORKDIR /app/instauto
+
+COPY package.json /app/instauto
+
+RUN yarn add instauto
+
```
Comment (mifi) [inline Dockerfile:11]:
```suggestion
```

File: Dockerfile (hunk 4 above)
Comment (mifi) [inline Dockerfile:14]:
```suggestion
CMD ["yarn"]
```

**Comment by mifi** [comment]
if these changes look ok then we can merge

File: example.js [hunk 7]
```diff
@@ -76,7 +76,9 @@ const options = {
     if (unfollowedCount > 0) await instauto.sleep(10 * 60 * 1000);
 
     // List of usernames that we should follow the followers of, can be celebrities etc.
-    const usersToFollowFollowersOf = ['lostleblanc', 'sam_kolder'];
+    const usersToFollowFollowersOf = process.env.USERS_TO_FOLLOW.length === 1 ?
+        process.env.USERS_TO_FOLLOW :
+        process.env.USERS_TO_FOLLOW.split(',');
```
Comment (mifi) [inline example.js:81]:
```suggestion
    const usersToFollowFollowersOf = process.env.USERS_TO_FOLLOW != null ? process.env.USERS_TO_FOLLOW.split(',') : [];
```

**Comment by mifi** [comment]
I haven't tested this but let me know if it doesnt work and we can do another PR to fix it

**Pull request merged**